### PR TITLE
Move d3 to a YUI module

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by makyo** [comment]
# QA

Mechanical change.  General QA with deploying services, bundles, placing units, etc.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2593/
Test PASSed.

File: app/assets/javascripts/d3-wrapper-end.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+return d3;
+  })();
+}, '0.1.0', {});
```
Comment (hatched) [inline app/assets/javascripts/d3-wrapper-end.js]: I'm not 100% sure on this, but for convoy to pick it up this last line might need to be

```
}, '0.1.0', {
  requires: []
}); 
```

Comment (mitechie) [inline app/assets/javascripts/d3-wrapper-end.js]: I think so as well

File: app/assets/javascripts/d3.status.js [hunk 2]
```diff
@@ -26,7 +26,9 @@
 YUI.add('d3-statusbar', function(Y) {
 
   var views = Y.namespace('juju.views'),
-      utils = Y.namespace('juju.views.utils');
+      utils = Y.namespace('juju.views.utils'),
+      ns = Y.namespace('d3'),
+      d3 = ns.d3;
```
Comment (mitechie) [inline app/assets/javascripts/d3.status.js]: so this is Y.d3.d3?

File: app/views/topology/bundle.js [hunk 3]
```diff
@@ -32,6 +32,7 @@ YUI.add('juju-view-bundle', function(Y) {
       utils = Y.namespace('juju.views.utils'),
       models = Y.namespace('juju.models'),
       d3ns = Y.namespace('d3'),
+      d3 = d3ns.d3,
```
Comment (mitechie) [inline app/views/topology/bundle.js]: and then same here.

**Comment by mitechie** [comment]
No QA atm, but I think the namespace is an extra level too deep. Can you let me know if I'm mistaken?

File: app/assets/javascripts/d3.status.js (hunk 2 above)
Comment (frankban) [inline app/assets/javascripts/d3.status.js]: +1

**Comment by frankban** [comment]
:+1: will QA after the "require" and namespace changes.
Thank you!

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2599/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2604/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2605/
Test FAILed.

**Comment by makyo** [comment]
Appears spurious, retrying http://ci.jujugui.org:8080/job/juju-gui/2607/

**Comment by makyo** [comment]
:shipit: per discussion on the call

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**